### ISSUE TITLE
Thesis Meeting | Oct 1, 2024 - 2:30pm - *Teams*

### ISSUE THREAD

**Issue by samm82** [samm82/TestGen-Thesis#98]
I'm sick once again so I'll be virtual for this meeting. Sorry for the once-again late notice! I like waiting to make an agenda until I know what the meeting will entail so that all relevant information gets contained in the email that gets sent out, but unfortunately that tends to happen a little later than ideal.

# Agenda
_This is a pretty loosely structured agenda, since most discussion work really just constitutes reviewing work I've done, so if there's something you two would like to focus on, I'm happy to pivot!_ 

1. Review of Introduction section #96 - any updates/initial notes?
2. Reclassification of discrepancies #97: Chapter 5 of my [thesis](https://github.com/samm82/TestGen-Thesis/blob/129c65b03818f6413aedf6c6288d5d465371229c/thesis.pdf) and Section IV of my [paper](https://github.com/samm82/TestGen-Thesis/blob/129c65b03818f6413aedf6c6288d5d465371229c/paper.pdf)
   _This probably makes more sense as a review issue, but we could start this process as a group discussion? I'm pretty sure this could be done in parallel with other section reviews…_
   1. Clarity of Terminology and Source Discrepancies
   2. Good choices for examples included in paper?
   3. Potential decomposition of discrepancies into finer categories (e.g., Definitions could be incorrect, missing, inconsistent, ambiguous, or overlapping)
      - It was brought up in #92 that defining these terms would be useful; is that only true if we end up using them? Is my current description of Definition Discrepancies too vague without defining these terms?

**Comment by samm82** [comment]
We discussed the introduction of orthogonal classifications of discrepancies: the "semantics" of the discrepancy (what I currently have: definitions, synonyms, etc.) and the "syntax" of the discrepancy (mistakes, omissions, inconsistencies, ambiguities or overlaps). These can be presented as two views of the discrepancies, so each discrepancy appears in _both_, as well as the third view of "subset discrepancies" (e.g., functional testing, correctness testing). This could also allow for these discrepancies to be "pulled out" more formally, such as into CSV files for individual processing. How this is done should also be outlined in the Tools section.

As an overarching thread, the methods for this should be formalized. For example, each source defines a set of terms (and a set of relations) that can be analyzed: do they use the same acronym twice? This should probably start from the ideal situation; for example, what should an acronym be?

# TODO
- [ ] Don't be clever with reducing duplication in terms, etc. (e.g., "A X and B X" instead of "A and B X" for clarity)
- [ ] Add "view" headers for discrepancies: Syntax, Semantics, and Subsets
- [ ] Regroup discrepancies in terms of "syntax", including its own table (akin to the 'Semantics' tables)
- [ ] Rename "discrepancy", since these don't always involve two conflicting sources
- [ ] Rename "Source Discrepancies" to "Citation Discrepancies"?
- [ ] Split out the subset discrepancies so they can be displayed in the other two views as well
- [ ] Update description of Tools section to accommodate the "two views" model

**Comment by samm82** [comment]
Since everything is addressed/moved to another issue, this can be closed
